portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java

```diff
@@ -56,12 +79,30 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     List<Comment> commentList =
         commentEntityList.stream()
             .map(entity
-                -> new Comment((String) entity.getProperty("username"),
-                    (String) entity.getProperty("commentText"), entity.getKey().getId()))
+                -> new Comment((String) entity.getProperty(USERNAME_ENTITY_PROPERTY),
+                    (String) entity.getProperty(COMMENT_TEXT_ENTITY_PROPERTY), entity.getKey().getId()))
             .collect(toList());
 
     response.setContentType("application/json");
     String json = new Gson().toJson(commentList);
     response.getWriter().println(json);
   }
+
+  @Override
+  public void doDelete(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      Query query = new Query(COMMENT_ENTITY);
+
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      PreparedQuery results = datastore.prepare(query);
+
+      List<Key> commentKeys = new ArrayList<>();
+      for (Entity entity : results.asIterable()) {
+        long id = entity.getKey().getId();
+        commentKeys.add(KeyFactory.createKey(COMMENT_ENTITY, id));
+      }
```

Comment:
Could inline this all (if you wanted to)

```suggestion
      results.asIterable().forEach(entity -> 
           commentKeys.add(KeyFactory.createKey(COMMENT_ENTITY, entity.getKey().getId())));
```

Reply:
Inlined